src/opnsense/mvc/app/controllers/OPNsense/Firewall/Api/AliasUtilController.php

```diff
@@ -90,14 +90,47 @@ public function aliasesAction()
      * list alias table
      * @param string $alias name to list
      * @return array alias contents
+     * @throws \Exception
      */
     public function listAction($alias)
     {
         $this->sessionClose();
+
+        $itemsPerPage = intval($this->request->getPost('rowCount', 'int', 9999));
+        $currentPage = intval($this->request->getPost('current', 'int', 1));
+        $offset = ($currentPage - 1) * $itemsPerPage;
+
         $backend = new Backend();
         $entries = json_decode($backend->configdpRun("filter list table", array($alias, "json")));
```

Comment:
Good to check with a large list, I kind of expect the configd call with > 1 Million results will still work like a charm, but if it doesn't we might need another approach here.  Did you test with a very large alias? and what was the reaction time in the browser for the api call.

Reply:
I tried it with the same alias as René, i.e. one with all Asian and African countries. That amounts to 61461 entries. The reaction time is indistinguishable from that of the bogons alias.